### PR TITLE
Added Form Component 

### PR DESCRIPTION
**Pull request by sankethgunasekara**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,5 @@
 
 # MSBuild Binary and Structured Log
 *.binlog
+*.json
```
Comment (tilangapramithvidusara) [inline .gitignore:18]: remove *.json. and put specific path. otherwise every json files are get ingnore

File: package.json [hunk 2]
```diff
@@ -16,15 +16,17 @@
     "refreshTypes": "pcf-scripts refreshTypes"
   },
   "dependencies": {
-    "@mui/icons-material": "^5.15.0",
+    "@mui/icons-material": "^5.14.18",
     "@mui/material": "^5.14.6",
     "@reduxjs/toolkit": "^1.9.7",
     "@types/to-json-schema": "^0.2.1",
+    "ajv": "^8.12.0",
```
Comment (tilangapramithvidusara) [inline package.json:23]: what is the matter using ajv

File: package.json [hunk 3]
```diff
@@ -16,15 +16,17 @@
...
     "@mui/material": "^5.14.6",
     "@reduxjs/toolkit": "^1.9.7",
     "@types/to-json-schema": "^0.2.1",
+    "ajv": "^8.12.0",
     "antd": "^5.8.4",
     "axios": "^1.5.1",
     "material-react-table": "^1.15.1",
     "patch-package": "^8.0.0",
-    "react-dom": "^17.0.2",
+    "react": "^18.2.0",
```
Comment (tilangapramithvidusara) [inline package.json:28]: why are you install react seperatly

File: .gitignore (hunk 1 above)
Comment (sankethgunasekara) [reply .gitignore:18]: got it

File: package.json (hunk 2 above)
Comment (sankethgunasekara) [reply package.json:23]: This was due to the dependency issue that I face during the "npm install"

File: package.json (hunk 3 above)
Comment (sankethgunasekara) [reply package.json:28]: This is also due to the dependency issue when installing the node modules. When I installed react separately only my PCF worked.

File: .gitignore [hunk 4]
```diff
@@ -15,3 +15,4 @@
...
```
Comment (tilangapramithvidusara) [inline .gitignore:18]: remove this *.json also

File: seerDiscoverDataTableROMPCF/src/components/Table/DyanamicTable.tsx [hunk 5]
```diff
@@ -1,67 +1,104 @@
...
-    if (info?.isDropDown) {
-      dispatch(setStateSnapshot(true))
-    }
-    // example for checking
-    // dispatch(setStateSnapshot(true))
-  }, [dispatch]);
+  const onChangeHanlder = useCallback(
+    (info) => {
+      dispatch(setSettingParameterAttributes(info));
+      if (info?.isDropDown) {
```
Comment (tilangapramithvidusara) [inline seerDiscoverDataTableROMPCF/src/components/Table/DyanamicTable.tsx:43]: also you can remove this if condition and statement

**Review by tilangapramithvidusara — APPROVED** [review]

**Pull request merged**